### DISCUSSION TITLE
Listen to  Tab's onMouseDown event outside the component

### DISCUSSION THREAD

**Discussion by dimitrisnl** [adobe/react-spectrum#2435] [Q&A] [answered]
Hello,

I've implemented a Tabs component using `useTab` & `useTabList`. I'm very happy with the finished product, so thank you for all the utilities that saved me tons of time.

I noticed that somewhere in my application there is a hook that listens to `onMouseDown` events and does some actions. Specifically, it closes some sidebars, but that's not relevant.

I noticed that the `TabList` component when clicked never makes that `eventListener` trigger.

Browsing through the code I noticed this snippet

https://github.com/adobe/react-spectrum/blob/2a0f664534de576201ef1251d6982d83bbcd01d8/packages/%40react-aria/interactions/src/usePress.ts#L333-L348

So my understanding, is that this event will never bubble up. I might be completely wrong, so feel free to correct me.

Is there a way I can make this interaction work? Having my `eventListener` trigger when I click/mousedown on a specific Tab? 

Thanks!

**Comment by dimitrisnl** [comment, marked as answer]
I have this workaround

```ts
  const onClick = (event: React.MouseEvent<HTMLElement>) => {
    if (typeof tabProps.onClick === 'function') {
      tabProps.onClick(event);

      const clonedNativeEvent = new Event('mousedown', event.nativeEvent);
      event.target.dispatchEvent(clonedNativeEvent);
    }
  };

  const nextTabProps = {...tabProps, onClick};
```

This is very suboptimal, and I understand that the consumer should normally use either an overlay to capture the clicks, or proper focus management.

**Reply by snowystinger** [reply]
I think you're saying your side bar is like a modal? I'm guessing that based on
> it closes some sidebars

and 
> the consumer should normally use either an overlay to capture the clicks

If that's correct, we handle it currently through useInteractOutside with capturing listeners https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/useInteractOutside.ts#L68
An overlay is another way to go, as you mentioned, and it's a good one if you have something that is acting as a modal.
We have hooks to aid you with focus management in modals as well, check out https://react-spectrum.adobe.com/react-aria/FocusScope.html

**Reply by dimitrisnl** [reply]
As it stands, the rest of the app isn't fully using react-aria. So the sidebar (not modal) isn't using proper focus management. 

There is also a requirement to be able to select stuff on the background before closing (think of a checkbox filter, when the sidebar is open). So having that extra click isn't doable for me.

Anyhow, I think this issue is due to the conflicting goals of the library, and my app. So maybe me dispatching the event again, makes sense for now, until I can rework on the sidebar behavior.

**Reply by devongovett** [reply]
You could also try a capturing listener eg onClickCapture. That way it will run before it gets to the tabs component.

**Reply by dimitrisnl** [reply]
Very interesting, thank you for the idea @devongovett . I'll check it out.

I'll mark my answer as correct to close this discussion and move it out of the way.
Thank you too @snowystinger for the help!
